typing/typecore.ml

```diff
@@ -8076,14 +8393,25 @@ let report_error ~loc env = function
       Location.errorf ~loc ~sub
         "@[This application is complete, but surplus arguments were provided afterwards.@ \
          When passing or calling a local value, extra arguments are passed in a separate application.@]"
-  | Param_mode_mismatch ty ->
+  | Param_mode_mismatch (ty, mkind) ->
+      let mkind =
+        match mkind with
+        | `Locality -> "local"
+        | `Uniqueness -> "unique"
+        | `Linearity -> "once"
+      in
       Location.errorf ~loc
-        "@[This function has a local parameter, but was expected to have type:@ %a@]"
-        Printtyp.type_expr ty
-  | Uncurried_function_escapes ->
+        "@[This function has a %s parameter, but was expected to have type:@ %a@]"
+        mkind Printtyp.type_expr ty
+  | Uncurried_function_escapes e ->
       Location.errorf ~loc
-        "This function or one of its parameters escape their region@ \
+        "This function or one of its parameters %s @ \
          when it is partially applied"
+        (match e with
+        | `Locality -> "escape their region"
+        | `Uniqueness -> assert false
+        | `Linearity -> "violates once constraint"
+        )
```

Comment:
The would be easier to read if the string was let bound. "violates once constraint" is a bit hard to read. How about "will be used 'many' when then were expected to be used 'once'"?